design/metadata-handling.md

```diff
@@ -0,0 +1,286 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Generating Metadata per node in CAPM3 down to Ironic
+
+## Status
+
+provisional
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a
+design and for highlighting any additional information provided beyond
+the standard template.
+
+[Tools for generating][] a table of contents from markdown are available.
+
+<!--ts-->
+   * [Title](#title)
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [User Stories [optional]](#user-stories-optional)
+            * [Story 1](#story-1)
+            * [Story 2](#story-2)
+         * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [Drawbacks [optional]](#drawbacks-optional)
+      * [Alternatives [optional]](#alternatives-optional)
+      * [References](#references)
+
+<!-- Added by: stack, at: 2019-02-15T11:41-05:00 -->
+
+<!--te-->
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+
+Cloud-init templates offer a very powerful feature to render the configuration
+at runtime based on metadata given by Ironic. We want to take advantage of this
```

Comment:
Where is Ironic going to provide that data?

Reply:
it will be given as metadata : https://cloudinit.readthedocs.io/en/latest/topics/instancedata.html It is accessible from within the userData template as `{{ ds.meta_data.<var_name> }}`